lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -0,0 +1,695 @@
+use crate::{BytesDecoder, StreamDecodingError};
+
+use super::{BoxedFramingError, FramingError};
+use bytes::{Buf, Bytes, BytesMut};
+use derivative::Derivative;
+use snafu::{ensure, Snafu};
+use std::any::Any;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use tokio;
+use tokio::task::JoinHandle;
+use tokio_util::codec::Decoder;
+use tracing::{info, warn};
+use vector_config::configurable_component;
+
+const GELF_MAGIC: &[u8] = &[0x1e, 0x0f];
+const GELF_MAX_TOTAL_CHUNKS: u8 = 128;
+const DEFAULT_TIMEOUT_SECS: f64 = 5.0;
+const DEFAULT_PENDING_MESSAGES_LIMIT: usize = 1000;
+
+const fn default_timeout_secs() -> f64 {
+    DEFAULT_TIMEOUT_SECS
+}
+
+const fn default_pending_messages_limit() -> usize {
+    DEFAULT_PENDING_MESSAGES_LIMIT
+}
+
+/// Config used to build a `ChunkedGelfDecoder`.
+#[configurable_component]
+#[derive(Debug, Clone, Default)]
+pub struct ChunkedGelfDecoderConfig {
+    /// Options for the chunked GELF decoder.
+    pub chunked_gelf: ChunkedGelfDecoderOptions,
+}
+
+impl ChunkedGelfDecoderConfig {
+    /// Build the `ChunkedGelfDecoder` from this configuration.
+    pub fn build(&self) -> ChunkedGelfDecoder {
+        ChunkedGelfDecoder::new(
+            self.chunked_gelf.timeout_secs,
+            self.chunked_gelf.pending_messages_limit,
+        )
+    }
+}
+
+/// Options for building a `ChunkedGelfDecoder`.
+#[configurable_component]
+#[derive(Clone, Debug, Derivative)]
+#[derivative(Default)]
+pub struct ChunkedGelfDecoderOptions {
+    /// The timeout, in seconds, for a message to be fully received. If the timeout is reached, the
+    /// decoder drops all the received chunks of the incomplete message and starts over.
+    #[serde(
+        default = "default_timeout_secs",
+        skip_serializing_if = "vector_core::serde::is_default"
```

Comment:
I don't know if this `skip_serializing_if` will work as intended

Here, the default is `default_timeout_secs`, which is 5s.
But `f64::default()` is 0s. 

As it is, this will be skipped only if `timeout_secs == 0`, right? 
I saw this pattern in https://github.com/vectordotdev/vector/blob/06ad674ff4fbf2df82c50b787c45316c5b5c8d50/lib/codecs/src/decoding/format/json.rs#L77

Reply:
Decided to remove the whole `skip_serializing_if` in b214fdb3e23352c5e353529280fda968909b1b5d. If you prefer another approach (for example, to test to the actual default value and skip then), tell me